### PR TITLE
Use a custom font across the app#45

### LINKED ISSUE #45: [Feat] Use a Custom Font Across the App
**Is your feature request related to a problem? Please describe.**
The current app design uses a default system font. While functional, it lacks a unique identity that sets the app apart and doesn't leverage a cohesive design aesthetic.

**Describe the solution you'd like**
Integrate a custom font throughout the app to improve the visual appeal and branding. The custom font should be applied globally across all text elements, ensuring consistency and uniqueness.

### PR DESCRIPTION
**Pull request by Sanat-Jha**
### Related Issue  
Closes #45 

### Type of Change
Put `x` inside the square bracket to specify what type of change your PR is:  
- [x] New Feature 

### Description of Change  
Added custom font (ubuntu from google fonts) across the whole app. Choice of font can be changed in main.dart.

### Implementation Details  
1. added google fonts in packages
2. updated theme data from 
 ` ` `
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
` ` ` 
to 
` ` ` 
theme: ThemeData(
        // Set the default font using Google Fonts
        textTheme: GoogleFonts.ubuntuTextTheme(
          Theme.of(context).textTheme,
        ),
        primarySwatch: Colors.blue,
      ),
` ` ` 

### Demo  
one can observe the new font applied
image

### CODE REVIEW

**Comment by pranavkonidena** [comment]
Please follow the PR format given at https://github.com/IMGIITRoorkee/MergeFest-Hacker/blob/main/CONTRIBUTORS.md

File: pubspec.lock [hunk 1]
```diff
@@ -204,9 +316,26 @@ packages:
...
+    description:
+      name: xdg_directories
+      sha256: "7a3f37b05d989967cdddcbb571f1ea834867ae2faa29725fd085180e0883aa15"
       url: "https://pub.dev"
     source: hosted
-    version: "13.0.0"
+    version: "1.1.0"
 sdks:
-  dart: ">=3.3.0 <4.0.0"
+  dart: ">=3.5.0 <4.0.0"
```
Comment (pranavkonidena) [inline pubspec.lock:340]: Please don't change flutter version as this may be a breaking change for others

Comment (pranavkonidena) [reply pubspec.lock:340]: U can downgrade ur version of flutter

File: lib/screens/home_screen.dart [hunk 2]
```diff
@@ -15,6 +16,7 @@ class HomeScreen extends StatelessWidget {
       floatingActionButton: FloatingActionButton(
```
Comment (pranavkonidena) [inline lib/screens/home_screen.dart:16]: This is a seperate issue, please tackle those in another PR

File: lib/screens/taskform_screen.dart [hunk 3]
```diff
@@ -23,6 +23,7 @@ class TaskFormScreen extends StatelessWidget {
             ElevatedButton(
               onPressed: () {
                 // Save task
+                Navigator.pop(context);
```
Comment (pranavkonidena) [inline lib/screens/taskform_screen.dart:26]: Same goes for this, this is a seperate PR. Please tackle this in another PR

**Review by pranavkonidena — CHANGES_REQUESTED** [review]

**Comment by Sanat-Jha** [comment]
changed fixed in latest commit

**Comment by pranavkonidena** [comment]
You have some merge conflicts which need to be resolved, please fix them

**Review by Akshit517 — CHANGES_REQUESTED** [review]
Instead of using google fonts for font , please include the .ttf font files directly in the project. This will avoid unnecessary dependencies

**Comment by sommye-ctr** [comment]
Since this PR is not yet merged & it seems the OP is working on other, can I pick this one? @Akshit517 @pranavkonidena

**Comment by pranavkonidena** [comment]
@sommye-ctr Please wait, if there is no update by tomorrow EOD, you can work on this

**Comment by pranavkonidena** [comment]
@Sanat-Jha please do the requisite changes by tomorrow EOD, it shouldn't take too much time for you

**Comment by Sanat-Jha** [comment]
I have done as asked by @Akshit517 to use ttf file in a new PR #54 .

**Pull request closed without merging**